### PR TITLE
Add review and design notes

### PR DESCRIPTION
**Pull request by stsquad**
Following the discussion on Monday document the desire for the separation of concerns even if we won't be there at the start.

### CODE REVIEW

**Comment by vireshk** [comment]
@stsquad please fix the failure 😀

**Review by vireshk — APPROVED** [review]

**Comment by andreeaflorescu** [comment]
@mathieupoirier can you also approve this PR as an ACK that you agree to becoming a maintainer? The process of adding maintainers is not yet documented, so at this point is tribal knowledge. I have an issue in my backlog to formally document it.

Once you approve this PR as well, I'll approve it, merge it, and add you both (@stsquad and @mathieupoirier) as maintainers.

**Review by mathieupoirier — APPROVED** [review]

**Review by andreeaflorescu — APPROVED** [review]

**Comment by andreeaflorescu** [comment]
@mathieupoirier @stsquad you should have an invitation pending to become maintainers of this repo.

**Pull request merged**
